sale_variant_configurator/models/sale_order.py

```diff
@@ -114,8 +114,34 @@ def _onchange_product_tmpl_id_configurator(self):
 
     @api.onchange('product_uom', 'product_uom_qty')
     def product_uom_change(self):
-        # TODO: Handle this when there's no product_id filled
-        super(SaleOrderLine, self).product_uom_change()
+        if self.product_id:
+            return super(SaleOrderLine, self).product_uom_change()
+        if self.product_tmpl_id and (self.order_id.pricelist_id and
+                                         self.order_id.partner_id):
+            product_tmpl = self.product_tmpl_id.with_context(
+                lang=self.order_id.partner_id.lang,
+                partner=self.order_id.partner_id.id,
+                quantity=self.product_uom_qty,
+                date_order=self.order_id.date_order,
+                pricelist=self.order_id.pricelist_id.id,
+                uom=self.product_uom.id,
+                fiscal_position=self.env.context.get('fiscal_position')
+            )
+            self.price_unit = self.env['account.tax']._fix_tax_included_price(
+                self._get_display_price(product_tmpl), product_tmpl.taxes_id,
+                self.tax_id)
+
+    @api.onchange('product_id')
+    def product_id_change(self):
+        if not self.product_id and self.product_tmpl_id:
```

Comment:
Why do you add this? The name is built automatically from template name + attributes from the base module. You are destroying here that names.